### PR TITLE
#93 support specifiying time zone by full name

### LINKED ISSUE #93: Feature request: support specifiying time zone by full name
`dateutil.parser.parse` can parse all the time formats acceptable to Postgres except for this one:

`2003-04-12 04:05:06 America/New_York`

(from http://www.postgresql.org/docs/9.4/static/datatype-datetime.html#DATATYPE-DATETIME-TIME-TABLE)

```python
>>> import dateutil.parser
>>> dateutil.parser.parse('2003-04-12 04:05:06 America/New_York')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/viktor/.virtualenvs/rewrite/lib/python3.4/site-packages/dateutil/parser.py", line 748, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
  File "/home/viktor/.virtualenvs/rewrite/lib/python3.4/site-packages/dateutil/parser.py", line 310, in parse
    res, skipped_tokens = self._parse(timestr, **kwargs)
TypeError: 'NoneType' object is not iterable

```

I tried passing a dictionary for the `tzinfos` parameter with a key for `'America/New_York'`, but it fails with the same error. I believe it fails because of the non-ASCII characters (/ and _) in the time zone.

### PR DESCRIPTION
**Pull request by aschatten**
@pganssle @jarondl
Fixes #93

### CODE REVIEW

File: dateutil/tz/tz.py [hunk 1]
```diff
@@ -8,12 +8,20 @@
...
+
+import re
 import struct
 import time
 import sys
 import os
 
 from six import string_types, PY3
+if PY3:
+    ifilter = filter
```
Comment (jarondl) [inline dateutil/tz/tz.py]: Please use `six.moves.filter`

**Comment by jarondl** [comment]
@aschatten, Why was this closed? Should I continue to review it?

**Comment by aschatten** [comment]
@jarondl I am going to recreate this PR. I got some messup with my local history, I am going to fix it now and re-create the PR.

**Comment by aschatten** [comment]
@jarondl Please review https://github.com/dateutil/dateutil/pull/153 instead of this one. Thanks.

**Pull request closed without merging**